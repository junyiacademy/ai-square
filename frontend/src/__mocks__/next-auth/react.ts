export const useSession = jest.fn(() => ({
  data: null,
  status: 'unauthenticated'
}));

export const signIn = jest.fn();
export const signOut = jest.fn();
export const getCsrfToken = jest.fn();
export const getSession = jest.fn();
export const getProviders = jest.fn();

export const SessionProvider = ({ children }: { children: React.ReactNode }) => children;