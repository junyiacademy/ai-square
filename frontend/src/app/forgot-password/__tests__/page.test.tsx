/**
 * Tests for page
 */

import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Page from '../page';

// Mock i18n module - need to match what ForgotPasswordClient expects
jest.mock('@/i18n', () => ({
  __esModule: true,
  default: {
    hasResourceBundle: jest.fn(() => true),
    loadNamespaces: jest.fn(),
    language: 'en'
  }
}));

// Mock react-i18next
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => key,
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    }
  }),
}));

// Mock dependencies
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    back: jest.fn(),
    refresh: jest.fn(),
  }),
  usePathname: () => '/',
  useSearchParams: () => new URLSearchParams(),
}));

describe('page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render without crashing', () => {
    const { container } = render(<Page />);
    expect(container).toBeInTheDocument();
  });

  it('should have proper structure', () => {
    render(<Page />);
    
    // Check for the title and form elements
    expect(screen.getByText('forgotPassword.title')).toBeInTheDocument();
    expect(screen.getByText('forgotPassword.subtitle')).toBeInTheDocument();
    
    // Check for email input
    const emailInput = screen.getByRole('textbox');
    expect(emailInput).toBeInTheDocument();
    expect(emailInput).toHaveAttribute('type', 'email');
    
    // Check for submit button
    const submitButton = screen.getByRole('button');
    expect(submitButton).toBeInTheDocument();
    
    // Check for back to login link
    const backLink = screen.getByRole('link');
    expect(backLink).toBeInTheDocument();
    expect(backLink).toHaveAttribute('href', '/login');
  });

  it('should handle form submission', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve({ success: true }),
      } as Response)
    );
    
    render(<Page />);
    
    const emailInput = screen.getByRole('textbox');
    const submitButton = screen.getByRole('button');
    
    // Enter email
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    expect(emailInput).toHaveValue('test@example.com');
    
    // Submit form
    fireEvent.click(submitButton);
    
    // Wait for success message
    await waitFor(() => {
      expect(screen.getByText('forgotPassword.successMessage')).toBeInTheDocument();
    });
    
    // Check that fetch was called
    expect(global.fetch).toHaveBeenCalledWith('/api/auth/forgot-password', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ email: 'test@example.com' }),
    });
  });

  it('should handle submission error', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve({ success: false, error: 'User not found' }),
      } as Response)
    );
    
    render(<Page />);
    
    const emailInput = screen.getByRole('textbox');
    const submitButton = screen.getByRole('button');
    
    // Enter email and submit
    fireEvent.change(emailInput, { target: { value: 'notfound@example.com' } });
    fireEvent.click(submitButton);
    
    // Wait for error message
    await waitFor(() => {
      expect(screen.getByText('User not found')).toBeInTheDocument();
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container.firstChild).toMatchSnapshot();
  });
});