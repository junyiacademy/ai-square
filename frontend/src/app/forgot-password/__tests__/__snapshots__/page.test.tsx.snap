// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`page should match snapshot 1`] = `
<div
  class="min-h-screen flex items-center justify-center bg-gradient-to-br from-blue-50 to-indigo-100 dark:from-slate-900 dark:to-indigo-900 py-12 px-4 sm:px-6 lg:px-8"
>
  <div
    class="max-w-md w-full space-y-8"
  >
    <div>
      <h2
        class="mt-6 text-center text-3xl font-extrabold text-gray-900 dark:text-white"
      >
        forgotPassword.title
      </h2>
      <p
        class="mt-2 text-center text-sm text-gray-600 dark:text-gray-400"
      >
        forgotPassword.subtitle
      </p>
    </div>
    <form
      class="mt-8 space-y-6"
    >
      <div>
        <label
          class="block text-sm font-medium text-gray-700 dark:text-gray-300"
          for="email"
        >
          email
        </label>
        <input
          autocomplete="email"
          class="mt-1 appearance-none relative block w-full px-3 py-2 border border-gray-300 dark:border-gray-600 placeholder-gray-500 dark:placeholder-gray-400 text-gray-900 dark:text-white rounded-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm dark:bg-gray-700"
          id="email"
          name="email"
          placeholder="emailPlaceholder"
          required=""
          type="email"
          value=""
        />
      </div>
      <div>
        <button
          class="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 disabled:opacity-50 disabled:cursor-not-allowed"
          type="submit"
        >
          forgotPassword.sendResetLink
        </button>
      </div>
      <div
        class="text-center"
      >
        <a
          class="text-sm text-indigo-600 hover:text-indigo-500 dark:text-indigo-400"
          href="/login"
        >
          forgotPassword.backToLogin
        </a>
      </div>
    </form>
  </div>
</div>
`;
